nvflare/app_opt/pt/fedopt.py

```diff
@@ -164,14 +166,17 @@ def server_update(self, model_diff):
 
         # Apply the update to the model. We must multiply weights_delta by -1.0 to
         # view it as a gradient that should be applied to the server_optimizer.
+        updated_params = []
         for name, param in self.model.named_parameters():
-            param.grad = torch.tensor(-1.0 * model_diff[name]).to(self.device)
+            if name in model_diff:
```

Comment:
Did we have some special handles for the server model initialization? It looks like `self.model.named_parameters() `does not contain unnamed parameters.

Reply:
No, this is specific to PyTorch. named_parameters only contains the trainable parameters while the state_dict (from which we get `model_diff`) also contains other parameters such as batch norm statistics.

Reply:
Thanks for the clarification!